src/main/java/org/jabref/gui/entryeditor/LatexCitationsTabViewModel.java

```diff
@@ -77,6 +85,16 @@ public void init(BibEntry entry) {
         currentEntry = entry;
         Optional<String> citeKey = entry.getCitationKey();
 
+        // make sure that we only add the listener once at the beginning
+        if (!hasListener) {
```

Comment:
In which cases can it happen that there is already a listener attached? Shouldn't the old one be removed when `init` is called a second time?

Reply:
Maybe, this branch is never executed? - Maybe, one needs to think more when to add the file listener. Typically, one has a method adding the listener. Then, this method is called twice: Once, when the view model is generatd and second at the change of the directory. At the change of a directory, the old listener should be removed.

Reply:
I was thinking of the scenario with the refresh button since that would call init with the same directory so I wanted to avoid adding duplicate listeners for the same path (not sure if that is how the listeners work though).

But I agree, the listener handling can be broken out into a method instead. Thanks for the review!